plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts

```diff
@@ -0,0 +1,245 @@
+ * using nunjucks jinja compat mode. Since this method mutates
+ * the global nunjucks instance, we can't enable this per-template,
+ * or only for templates with cookiecutter compat enabled, so the
+ * next best option is to explicitly enable it globally and allow
+ * folks to rely on jinja compatibility behaviour in fetch:template
+ * templates if they wish.
+ *
+ * cf. https://mozilla.github.io/nunjucks/api.html#installjinjacompat
+ */
+nunjucks.installJinjaCompat();
+
+export type FetchTemplateInput = {
+  url: string;
+  targetPath?: string;
+  values: any;
+  copyWithoutRender?: string[];
+  cookiecutterCompat?: boolean;
+};
+
+export function createFetchTemplateAction(options: {
+  reader: UrlReader;
+  integrations: ScmIntegrations;
+}) {
+  const { reader, integrations } = options;
+
+  return createTemplateAction<FetchTemplateInput>({
+    id: 'fetch:template',
+    description:
+      "Downloads a skeleton and will template variables into the skeleton and places the result in the workspace, or optionally in a subdirectory specified by the 'targetPath' input option.",
+    schema: {
+      input: {
+        type: 'object',
+        required: ['url'],
+        properties: {
+          url: {
+            title: 'Fetch URL',
+            description:
+              'Relative path or absolute URL pointing to the directory tree to fetch',
+            type: 'string',
+          },
+          targetPath: {
+            title: 'Target Path',
+            description:
+              'Target path within the working directory to download the contents to. Defaults to the working directory root.',
+            type: 'string',
+          },
+          values: {
+            title: 'Template Values',
+            description: 'Values to pass on to the templating engine',
+            type: 'object',
+          },
+          copyWithoutRender: {
+            title: 'Copy Without Render',
+            description:
+              'An array of glob patterns. Any files or directories which match are copied without being processed as templates.',
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+          cookiecutterCompat: {
+            title: 'Cookiecutter compatibility mode',
+            // TODO(mtlewis): documentation for cookiecutter compat mode
+            description:
+              'Enable features to maximise compatibility with templates built for fetch:cookiecutter',
+            type: 'boolean',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      ctx.logger.info('Fetching template content from remote URL');
+
+      const workDir = await ctx.createTemporaryDirectory();
+      const templateDir = resolvePath(workDir, 'template');
+
+      const targetPath = ctx.input.targetPath ?? './';
+      const outputDir = path.resolve(ctx.workspacePath, targetPath);
+
+      if (!outputDir.startsWith(ctx.workspacePath)) {
```

Comment:
Can we use [resolvedSafeChildPath](https://github.com/backstage/backstage/pull/6204) here?

Reply:
Looking into this!

Reply:
Done in c22fc45ff